core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/openapi.kt

```diff
@@ -0,0 +1,1464 @@
+        readOpenApiAsString(file.readText(), extensionProperties = extensionProperties)
+
+    override fun acceptsExtension(ext: String): Boolean = ext in listOf("yaml", "yml", "json")
+
+    override val testOrder: Int = 60000
+
+    override fun createDefaultReadMethod(pathRepresentation: String?): DefaultReadDfMethod = DefaultReadOpenApiMethod
+}
+
+/**
+ * Attempt to undo the creation of `value` and `array` columns by the json -> DF decoder.
+ */
+private fun DataRow<*>.unwrapJsonColumn(): Any? {
+    // for clashes like array and array1, take the highest number (this may break, but not much I can do about it)
+    val valueName = columnNames().filter { it.contains(valueColumnName) }.maxOfOrNull { it }
+    val arrayName = columnNames().filter { it.contains(arrayColumnName) }.maxOfOrNull { it }
+
+    return when {
+        getVisibleValues().isEmpty() -> null
+
+        // Can't unwrap anything
+        valueName == null && arrayName == null -> this
+
+        else -> valueName?.let(::getOrNull) ?: arrayName?.let(::getOrNull)
+    }
+}
+
+/**
+ * Attempt to undo the creation of `value` and `array` columns by the json -> DF decoder for Maps.
+ */
+@Suppress("UNCHECKED_CAST")
+private fun Map<String, *>.unwrapJsonColumn(): Map<String, Any?> =
+    mapValues { (_, value) ->
+        when {
+            value is DataRow<*>? -> value?.unwrapJsonColumn()
+
+            value is Map<*, *> && value.keys.all { it is String } ->
+                (value as Map<String, *>).unwrapJsonColumn()
+
+            else -> value
+        }
+    }
+
+/**
+ * Function to be used in [ConvertSchemaDsl] ([AnyFrame.convertTo]) to help convert a DataFrame to adhere to an
+ * OpenApi schema.
+ */
+@Suppress("RemoveExplicitTypeArguments")
+public fun ConvertSchemaDsl<*>.convertDataRowsWithOpenApi() {
+    // undo Json wrapping of values that should be Any?
+    convert<DataRow<*>>().with<_, Any?>(DataRow<*>::unwrapJsonColumn)
+
+    // undo Json wrapping of maps with values that should be Any?
+    convert(
+        from = { it == typeOf<DataRow<*>>() },
+        to = { // any type of Map<String, Any? / *> or Map<String, Any? / *>?
+            (it.isSubtypeOf(typeOf<Map<*, *>>()) || it.isSubtypeOf(typeOf<Map<*, *>?>())) &&
+                it.arguments.getOrNull(0)?.type == typeOf<String>() &&
+                it.arguments.getOrNull(1)?.type.let { it == typeOf<Any?>() || it == null }
+        }
+    ) { (it as DataRow<*>).toMap().unwrapJsonColumn() }
+
+    // convert DataRows to Maps if required by the schema
+    convert(
+        from = { it == typeOf<DataRow<*>>() },
+        to = { // any type of Map<String, _> or Map<String, _>?
+            (it.isSubtypeOf(typeOf<Map<*, *>>()) || it.isSubtypeOf(typeOf<Map<*, *>?>())) &&
+                it.arguments.getOrNull(0)?.type == typeOf<String>()
+        }
+    ) { (it as DataRow<*>).toMap() }
+
+    // convert DataFrame to DataFrame<Any?> if required by the schema
```

Comment:
What's the special meaning of DataFrame<Any?>? 
Previously, i think, you could just take any existing dataframe would be a valid value of type DataFrame<Any?>. But here's converting

Reply:
ok, it got commented out :D

Reply:
This is not present anymore in later commits :)